lib/Sema/ConstraintSystem.h

```diff
@@ -3009,6 +3024,59 @@ class DisjunctionChoice {
     return decl->isOperator() ? decl : nullptr;
   }
 };
+
+/// \brief Constraint "bucket" represents a single
+/// "solvable" unit of the constraint system which
+/// could be split further into more distinct units.
```

Comment:
I don't think this statement is clear. I think I know what you mean here, but it might be clearer to point out that it's a single solvable unit, but the process of assigning types in some cases allows it to be further split.